src/services/slack.ts

```diff
@@ -52,25 +53,49 @@ export class SlackConnector implements ISlackConnector {
     }
     return {};
   }
-  parseSelection(stateValues: any): any {
+
+  async parseSelection(
+    stateValues: any,
+    isAdmin: boolean,
+    repoBranchesRepository: RepoBranchesRepository
+  ): Promise<any> {
     const values = {};
     const inputMapping = {
       block_repo_option: 'repo_option',
       block_hash_option: 'hash_option',
+      block_deploy_option: 'deploy_option',
     };
 
+    // if deploy all was selected:
+    if (stateValues['block_deploy_option']['deploy_option']?.selected_option?.value == 'deploy_all') {
+      if (isAdmin) {
+        values['deploy_option'] = 'deploy_all';
+        values['repo_option'] = await repoBranchesRepository.getProdDeployableRepoBranches();
+        return values;
+      } else
+        throw {
```

Comment:
We should `return` instead of `throw` here. If we throw, the API response will not be this because API gateway expects an object to be returned from the Lambda function.

Reply:
Following up on this, if we do want to throw here, let's create a new error object and throw that e.g. `throw new Error('error occurred');`. In the handler function where we catch the error, we can then return the object that you were originally throwing. If you need to differentiate between error types, you can either use the message to determine that, or maybe add an error type property. The latter approach would be a bit more work, so don't worry about it too much.

The benefit of this is that it's clear what the handler function will be returning.

Reply:
Thank you for the explanation! I've made changes so that an error object is thrown. How does that look to you?